### ISSUE TITLE
UDOP pretraining with MAE decoder

### ISSUE THREAD

**Issue by haixpham** [microsoft/i-Code#73]
I am attempting to reproduce the original UDOP pretraining code as in paper. I have question - is the image reconstruction loss optimized together with text generation loss within the same batch or are they optimized alternatingly? This is because in UDOP model forward() pass, either image loss or text loss is returned depending on ids_keep is None (text) or not (image), but not both at the same time, so I wonder which approach was used in the original code.

TIA

**Comment by haixpham** [comment]
Please ignore my question, I found the answer. At the beginning of forward() pass this loop
```
if input_dict is not None:
            return_task_outputs = []
            for task in input_dict:
                return_task_outputs.append(self.forward(**input_dict[task]))
            return return_task_outputs
```
computes the loss for each training task separately, then they are summed up in trainer.training_step()

**Comment by ofir1080** [comment]
Hi @haixpham 
Can you please share you managed to reproduce the pretraining?
BTW, are you sure that the supplied checkpoints are pretraining? I saw that they used them for inference in RVL-CDIP for doc. classification.
Thanks!

**Comment by haixpham** [comment]
@ofir1080 Unfortunately this is for a company project so I'm not allowed to share code at the moment. The code in this repository is for downstream finetuning, the code for pretraining is at a different repo

**Comment by haixpham** [comment]
I'm not the author of the work. From their introduction, the weights are  result of pretraining
